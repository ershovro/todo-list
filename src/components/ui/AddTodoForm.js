import React from 'react';

const AddTodoForm = ({ oneNewTodo = f => f }) => {
   const _onSubmitHandler = (e) => {
      e.preventDefault();
      oneNewTodo(_input.value);
      _input.value = '';
      _input.focus();
   };
   let _input;

   return (
      <form onSubmit={_onSubmitHandler} className="addTodoForm">
         <input
            ref={input => _input = input}
            type="text"
            required
         />
         <input
            type="submit"
            value="Add todo"
         />
      </form>
   );
};


export default AddTodoForm;