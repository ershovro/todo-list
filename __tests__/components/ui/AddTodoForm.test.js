import React from 'react';
import AddTodoForm from "../../../src/components/ui/AddTodoForm";
import enzyme from 'enzyme';
import { JSDOM } from 'jsdom';
import toJSON from 'enzyme-to-json'
import { compose } from 'redux';
import Adapter from 'enzyme-adapter-react-16';


const { window } = new JSDOM(`...`);
const { document } = (new JSDOM(`...`)).window;

enzyme.configure({ adapter: new Adapter() });
global.window = window;
global.document = document;


describe('<AddTodoForm /> ui component', () => {

   describe('Rendering ui', () => {

      /*it('Renders Correctly', () => {
         compose(expect, toJSON, enzyme.shallow)(
            <AddTodoForm />
         ).toMatchSnapshot();
      });*/

      it('Renders form with class .addTodoForm', () => {
         expect(
            enzyme.mount( <AddTodoForm /> ).
            find('form.addTodoForm').
            length
         ).toBe(1)
      });

      it('Renders 1 form', () => {
         expect(
            enzyme.mount( <AddTodoForm /> ).
            find('form').
            length
         ).toBe(1)
      });

      it('Submitting default does not cause error', () => {
         const wrapper = enzyme.mount(  <AddTodoForm /> );

         wrapper.find('input').first().value = 'Wake up';
         wrapper.find('form').simulate('submit');
      });

   });

   describe('Adding new todo', () => {
      const _oneNewTodo = jest.fn();
      const wrapper = enzyme.mount(  <AddTodoForm oneNewTodo={_oneNewTodo}/> );

      wrapper.find('input[type="text"]').instance().value = 'Wake up';
      wrapper.find('form').simulate('submit');

      it('Invoke onSubmit handler', () => {
         expect(_oneNewTodo).toBeCalled();
      });

      it('Invoke onSubmit handler with correct arguments', () => {
         expect(_oneNewTodo).toBeCalledWith('Wake up');
      });

      it('Reset the text value', () => {
         expect( wrapper.find('[type="text"]').instance().value ).
         toBe('')
      });

   });
});
