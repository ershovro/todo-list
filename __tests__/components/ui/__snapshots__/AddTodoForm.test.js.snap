// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddTodoForm /> ui component Rendering ui Renders Correctly 1`] = `
<form
  className="addTodoForm"
  onSubmit={[Function]}
>
  <input
    required={true}
    type="text"
  />
  <input
    type="submit"
    value="Add todo"
  />
</form>
`;
